apps/system/init/init.c

```diff
@@ -157,5 +167,14 @@ int preapp_start(int argc, char *argv[])
 	}
 #endif
 
+/***********************************************************************************
+ *	current preapp_start does the up_cxxinitialize which initializes the
+ *	static constructors. All the tasks and threads created by preapp refer to
+ *	these constructors. To make sure that c++ test cases from tash runs properly,
+ *	we need to keep preapp/constructor initializations alive with below changes.
+ **********************************************************************************/
+#ifdef CONFIG_HAVE_CXXINITIALIZE
+		sigwaitinfo(&sig_set, NULL);
```

Comment:
As far as I know, signal can be awakened by another event like interrupt or something else. Could you confirm this is enough to keep alive?

Reply:
and need to check indentation

Reply:
and I don't know what is better than semaphore?

Reply:
Yes agree, signal can be awakened and I need to handle again to go back to wait.
We will use semaphore as it makes sure the count is locked always with the preapp and never get unlocked by other events.